Tests/AK/TestIPv4AddressCidr.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, famfo <famfo@famfo.xyz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <AK/IPv4Address.h>
+#include <AK/IpAddressCidr.h>
+
+TEST_CASE(sanity_check)
+{
+    auto address_result = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 32);
+
+    EXPECT(!address_result.is_error());
+
+    IPv4AddressCidr address = address_result.release_value();
+
+    EXPECT_EQ(address.length(), (u32)32);
+    EXPECT_EQ(address.ip_address(), IPv4Address(192, 0, 2, 1));
+    EXPECT_EQ(address.first_address_of_subnet(), IPv4Address(192, 0, 2, 1));
+    EXPECT_EQ(address.last_address_of_subnet(), IPv4Address(192, 0, 2, 1));
+    EXPECT_EQ(address.netmask(), IPv4Address(255, 255, 255, 255));
+    EXPECT(address.contains(IPv4Address(192, 0, 2, 1)));
+}
+
+TEST_CASE(should_fail_on_invalid_length)
+{
+    auto address_result = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 33);
+    EXPECT(address_result.is_error());
+    EXPECT_EQ(address_result.error(), IPv4AddressCidr::IPAddressCidrError::CidrTooLong);
+}
+
+TEST_CASE(should_find_first_in_subnet)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 24).release_value();
+    EXPECT_EQ(address.first_address_of_subnet(), IPv4Address(192, 0, 2, 0));
+}
+
+TEST_CASE(should_find_last_in_subnet)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 24).release_value();
+    EXPECT_EQ(address.last_address_of_subnet(), IPv4Address(192, 0, 2, 255));
+}
+
+TEST_CASE(should_return_matching_netask)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 24).release_value();
+    EXPECT_EQ(address.netmask(), IPv4Address(255, 255, 255, 0));
+}
+
+TEST_CASE(should_contain_other)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 24).release_value();
+    EXPECT(address.contains(IPv4Address(192, 0, 2, 100)));
+}
+
+TEST_CASE(should_set_address)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 8).release_value();
+    EXPECT_EQ(address.ip_address(), IPv4Address(192, 0, 2, 1));
+
+    address.set_ip_address(IPv4Address(198, 51, 100, 1));
+    EXPECT_EQ(address.ip_address(), IPv4Address(198, 51, 100, 1));
+}
+
+TEST_CASE(should_set_length)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 32).release_value();
+    EXPECT_EQ(address.length(), (u32)32);
+
+    EXPECT(!address.set_length(24).is_error());
+    EXPECT_EQ(address.length(), (u32)24);
+}
+
+TEST_CASE(should_not_set_invalid_length)
+{
+    IPv4AddressCidr address = IPv4AddressCidr::create(IPv4Address(192, 0, 2, 1), 32).release_value();
+    EXPECT(address.set_length(33).is_error());
+}
```

Comment:
Isn't this basically exactly the same test case as `should_fail_on_invalid_length`? Just checking to see if I'm missing some slight variation here that I may also miss on other test cases.

Reply:
It is essentially the same test case but I want to cover all code paths with the tests.

Reply:
Ah, whoops, I missed that `should_fail_on_invalid_length` sets the length through the constructor.